cpp/onedal/error_handling.hpp

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
```

Comment:
Need to think better about the name of the file. It does not contain "error handling" routines, just exception classes.
I think we should be aligned with C++ here and name it "exception.hpp", or event decompose it on many files like in C++ implementations.